game/BeepSaber_Game.gd

```diff
@@ -2,6 +2,16 @@
 #
 extends Spatial
 
+# TODO add better state management
```

Comment:
I should probably elaborate on this...
The code makes a bunch of stateful assumptions everywhere (ie. node visibility).
I started to introduce the concept of game states, but then never followed through with the full refactor.
Thoughts?

Reply:
Fine for me to keep it like that for now and address this in the future. But maybe you add the information from your github comment also in the comment inside the code; I think it makes it easier for others to understand if they see it in the future without the PR.